mock/world/mockESDTAccount.go

```diff
@@ -238,3 +242,70 @@ func (a *Account) GetTokenNames() ([][]byte, error) {
 
 	return tokenNames, nil
 }
+
+// MockESDTData groups together all instances of a token (same token name, different nonces).
+type MockESDTData struct {
+	TokenIdentifier []byte
+	Instances       []*esdt.ESDigitalToken
+	LastNonce       uint64
+	Roles           [][]byte
+}
+
+func getOrCreateMockESDTData(tokenName string, resultMap map[string]*MockESDTData) *MockESDTData {
+	resultObj := resultMap[tokenName]
+	if resultObj == nil {
+		resultObj = &MockESDTData{
+			TokenIdentifier: []byte(tokenName),
+			Instances:       nil,
+			LastNonce:       0,
+			Roles:           nil,
+		}
+		resultMap[tokenName] = resultObj
+	}
+	return resultObj
+}
+
+// GetFullMockESDTData returns the information about all the ESDT tokens held by the account.
+func (a *Account) GetFullMockESDTData() (map[string]*MockESDTData, error) {
+	resultMap := make(map[string]*MockESDTData)
+	for key := range a.Storage {
+		if IsTokenKey([]byte(key)) {
+			tokenInstance, err := a.GetTokenData([]byte(key))
+			if err != nil {
+				return nil, err
+			}
+
+			if tokenInstance.Value.Sign() > 0 {
+				var tokenName string
+				if tokenInstance.TokenMetaData == nil || tokenInstance.TokenMetaData.Nonce == 0 {
+					// ESDT, no nonce in the key
+					tokenNameFromKey := GetTokenNameFromKey([]byte(key))
+					tokenInstance.TokenMetaData = &esdt.MetaData{
+						Name:  tokenNameFromKey,
+						Nonce: 0,
+					}
+					tokenName = string(tokenNameFromKey)
+				} else {
+					// the key also contains the nonce, we take the token identifier from the metadata
+					tokenName = string(tokenInstance.TokenMetaData.Name)
+				}
```

Comment:
Replace with a function that takes `tokenInstance` and `key` and returns `tokenName`.